fendermint/vm/interpreter/src/fvm/checkpoint.rs

```diff
@@ -255,6 +255,12 @@ where
             .await
             .context("failed to broadcast checkpoint signature")?;
 
+            emit(CheckpointSigned {
+                height: height.value(),
+                hash: HexEncodableBlockHash(cp.block_hash.to_vec()),
+                validator: validator_ctx.public_key,
+            });
```

Comment:
Yes, exactly. I am aware this is not ideal since it won't publish other votes, but I thought having at least its own signature is better than nothing for now. However, your suggestion makes sense. I will also create a ticket to track it as a future update.